test/test_cli.rb

```diff
@@ -40,6 +40,30 @@ def test_invalid_options_ignored_recursive_current_dir
     assert_equal %w[.], options.files
   end
 
+  def test_options_file
+    option_file_path = File.expand_path('../fixtures/extra-options.txt', __FILE__)
+    option_file_path2 = File.expand_path('../fixtures/extra-options2.txt', __FILE__)
+    options = process_args([
+      '--exclude=1',
+      '--options', option_file_path,
+      '-R',
+      '--exclude=2',
+      "--options=#{option_file_path2}",
+      '--exclude=3',
+    ])
+    assert_equal true, options.recursive
+    assert_equal %w[.], options.files
+    assert_equal %w[.git 1 path\ with\ spaces bundle/with\ spaces/* 2 bundle 3], options.exclude
+    assert_equal 'emacs', options.format
+  end
+
+  def test_options_file_with_unsupported
```

Comment:
The only problem here: this test fails with `POSIXLY_CORRECT` set in env.

This is because OptionParser mimics [GNU getopt](https://www.gnu.org/software/libc/manual/html_node/Using-Getopt.html#Using-Getopt) [behavior] (https://github.com/ruby/ruby/blob/trunk/lib/optparse.rb#L1645). in this mode, option processing stops after the first non-option, which is `lib`. All following arguments are treated as arguments, even if they look like options.

I'd leave it as-is - this is as designed, having options in the wrong order is an error in POSIX, if anyone cares.

Another test with errors in this mode is `test_invalid_options_ignored_plus_files`, which again, stops at the first non-option and never sees `--ignore-unsupported` as an option. Again - I'd leave this in, same reasoning.